### PR TITLE
[modules] Fix some warnings and OglAttribute handleTopologyChange

### PR DESCRIPTION
**Pull request by epernod**
Various fix from my working branch:
- FIX: wrong msg_error instead of msg_info in PrecomputedConstraintCorrection
- FIX: warnings when drawing collision interaction using UnilateralInteractionConstraint
- FIX: OglAttribute handleTopologyChange method by using a PointData for value array.
- CLEAN: some logs in PointModel
- FIX: NearestPointROI to use rest_position instead of position

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaConstraint/PrecomputedConstraintCorrection.inl [hunk 1]
```diff
@@ -413,13 +413,13 @@ void PrecomputedConstraintCorrection<DataTypes>::bwdInit()
     //  Print 400 first row and column of the matrix
     if (this->notMuted())
     {
-        msg_error(this) << "Matrix compliance : nbCols = " << nbCols << "  nbRows =" << nbRows;
+        msg_info(this) << "Matrix compliance : nbCols = " << nbCols << "  nbRows =" << nbRows;
```
Comment (damienmarchal) [inline modules/SofaConstraint/PrecomputedConstraintCorrection.inl]: remove the "this"

File: modules/SofaConstraint/UnilateralInteractionConstraint.inl [hunk 2]
```diff
@@ -417,13 +417,11 @@ void UnilateralInteractionConstraint<DataTypes>::draw(const core::visual::Visual
         redVertices.push_back(c.P);
         redVertices.push_back(c.Q);
 
-        otherVertices.push_back(c.P);
-        otherColors.push_back(sofa::defaulttype::RGBAColor::white());
+        otherVertices.push_back(c.P);        
         otherVertices.push_back(c.P + c.norm);
-        otherColors.push_back(sofa::defaulttype::RGBAColor(0,0.5,0.5,1));
+        otherColors.push_back(sofa::defaulttype::RGBAColor::white());
```
Comment (damienmarchal) [inline modules/SofaConstraint/UnilateralInteractionConstraint.inl:422]: +1

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: modules/SofaGeneralEngine/NearestPointROI.inl [hunk 3]
```diff
@@ -101,8 +101,9 @@ void NearestPointROI<DataTypes>::doUpdate()
     indices2->clear();
 
     const Real maxR = f_radius.getValue();
-    const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::position())->getValue();
-    const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::position())->getValue();
+    
+    const VecCoord& x1 = this->mstate1->read(core::ConstVecCoordId::restPosition())->getValue();
+    const VecCoord& x2 = this->mstate2->read(core::ConstVecCoordId::restPosition())->getValue();
```
Comment (damienmarchal) [inline modules/SofaGeneralEngine/NearestPointROI.inl:106]: Sorry to ask late about htis but why do you change this ? 
Shouldn't it be selectable by the user ?

**Comment by damienmarchal** [comment]
I have added just one small question i forgot to ask during the meeting;

**Pull request merged**